crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -109,10 +109,16 @@ impl<'a> Format<PyFormatContext<'_>> for FormatString<'a> {
                 if is_implicit_concatenation(string_content) {
                     in_parentheses_only_group(&FormatStringContinuation::new(self.string)).fmt(f)
                 } else {
-                    FormatStringPart::new(string_range, self.string.quoting(&f.context().locator()))
-                        .fmt(f)
+                    FormatStringPart::new(
+                        string_range,
+                        self.string.quoting(&f.context().locator()),
+                        &f.context().locator(),
+                        f.options().quote_style(),
+                    )
+                    .fmt(f)
                 }
             }
+            StringLayout::DocString => format_docstring(self.string.range(), f),
```

Comment:
Nit: The `format_docstring` function is the odd one in this file. Most other code uses `Format` implementations.

Reply:
i thought about this too but i didn't find a design that makes this better; i'd rather turn `FormatStringContinuation` into a function call.

In general, what would you think about going more into the direction of trying to make function calls (or some indirection of them) format-able?

Reply:
You can return a format_with or format_once.